testing/regress/ecl/redislockingtest.ecl

```diff
@@ -0,0 +1,63 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2014 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+IMPORT * FROM lib_redis.locking;
+IMPORT FlushDB FROM lib_redis.sync;
+
+STRING servers := '--SERVER=127.0.0.1:6379';
+
+//SetBoolean(servers, 'b', TRUE);
+//GetBoolean(servers, 'b');
+
+UNSIGNED lock := GetLockObject(servers, 'pi');
+//GetLockObject(servers, 'pi');
+//GetLockObject(servers, 'pi');
+NOFOLD(MissThenLock(lock));
+NOFOLD(MissThenLock(lock));
+
+
+REAL pi := 3.14159265359;
```

Comment:
Yes, as I mentioned on flowdock, this isn't a test at all. This is the main part that I need help with regarding hoisting. The examples in the README file work so long as that is the _only_ redis plugin functions present due to hoisting. It has no key file. It is all commented out other than some unrelated and the two MissThenLock calls. Monitoring this from redis-cli showed that GetLockObject was being called twice rather than once.

I'll move it to the ecl regress dir for now.